server/repository/src/database_settings.rs

```diff
@@ -56,10 +64,64 @@ impl DatabaseSettings {
     }
 }
 
+// feature sqlite
+#[cfg(not(feature = "postgres"))]
+#[derive(Debug)]
+pub struct SqliteConnectionOptions {
+    pub enable_wal: bool,
+    pub enable_foreign_keys: bool,
+    pub busy_timeout_ms: Option<u32>,
+}
+// feature sqlite
+#[cfg(not(feature = "postgres"))]
+impl diesel::r2d2::CustomizeConnection<SqliteConnection, diesel::r2d2::Error>
+    for SqliteConnectionOptions
+{
+    fn on_acquire(&self, conn: &mut SqliteConnection) -> Result<(), diesel::r2d2::Error> {
+        (|| {
+            if self.enable_wal {
+                // println!("Adding WAL");
+                let _r =
+                    conn.batch_execute("PRAGMA journal_mode = WAL; PRAGMA synchronous = NORMAL;");
```

Comment:
doesn't `throw` an error (try adding something invalid after NORMAL;) . this is probably a good use case for `expect`, ` conn.batch_execute("PRAGMA journal_mode = WAL; PRAGMA synchronous = NORMAL;").expect("Failed to ..."); Also closure can be removed

Reply:
> Done, However I think there could be a very rare edge case where something in on_acquire() errors but otherwise the code could continue

Sure, i wasn't so much concerned with an error in statement, comment was mainly due to obsolete closure (that wasn't actually catching errors, as none were thrown).

Thanks for change (but still extra brackets)